grpc/src/main/java/com/linecorp/armeria/server/grpc/FramedGrpcService.java

```diff
@@ -342,7 +347,12 @@ public boolean isFramed() {
 
     @Override
     public List<ServerServiceDefinition> services() {
-        return registry.services();
+        return ImmutableList.copyOf(registry.services().values());
```

Comment:
How about caching this upfront so we don't have to create a new copy on every call? Alternatively, we could make `HandlerRegister.services()` to return `ImmutableList` and add an `instanceof ImmutableList` assertion.

Reply:
It looks good to return `ImmutableList`. :)